src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1817,8 +1826,99 @@ impl Cage {
                     } // Trigger SIGPIPE
                     retval
                 }
+                Stream(_stream_filedesc_obj) => {
+                    let iovecslice = concat_iovec_to_slice(iovec, iovcnt);
+                    log_from_slice(&iovecslice)
+                }
+                File(ref mut normalfile_filedesc_obj) => {
+                    // Check if the file is open for writing
+                    if is_rdonly(normalfile_filedesc_obj.flags) {
+                        return syscall_error(
+                            Errno::EBADF,
+                            "writev",
+                            "specified file not open for writing",
+                        );
+                    }
+                    // Retrieve the inode object for the file
+                    let mut inodeobj = FS_METADATA
+                        .inodetable
+                        .get_mut(&normalfile_filedesc_obj.inode)
+                        .unwrap();
+    
+                    match *inodeobj {
+                        Inode::File(ref mut normalfile_inode_obj) => {
+                            let position = normalfile_filedesc_obj.position;
+                            let filesize = normalfile_inode_obj.size;
+                            let blankbytecount = position as isize - filesize as isize;
+                            // Retrieve the file object from the file object table
+
+                            let mut fileobject = FILEOBJECTTABLE
+                                .get_mut(&normalfile_filedesc_obj.inode)
+                                .unwrap();
+    
+                            // We need to pad the file with blank bytes if we are at a position past the end of the file
+                            if blankbytecount > 0 {
+                                if let Ok(byteswritten) = fileobject.zerofill_at(filesize, blankbytecount as usize) {
+                                    if byteswritten != blankbytecount as usize {
+                                        panic!("Write of blank bytes for writev failed!");
+                                    }
+                                } else {
+                                    panic!("Write of blank bytes for writev failed!");
+                                }
+                            }
+                            // Create IoSlice objects from the raw iovec pointer
+                            let iovecs = unsafe { std::slice::from_raw_parts(iovec, iovcnt as usize) };
+                            let iovs: Vec<IoSlice> = iovecs.iter()
```

Comment:
do we know if this copies to new memory or does it just wrap the existing memory?